KAN-155 feat: 마이페이지 회원 탈퇴 기능 추가

dayaelee opened the pull request:
### 🔗 관련 이슈
https://ktb-22.atlassian.net/browse/KAN-155?atlOrigin=eyJpIjoiOTA3NzJmNjUxZGE0NGZmYjllOWM0Yjc3MWIyYjViZWIiLCJwIjoiaiJ9

### 📌 작업 내용

- 마이페이지 회원 탈퇴 기능 추가
- 회원 탈퇴 버튼 스타일 변경

### 🧑‍💻 작업 유형
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드 리팩토링
- [ ] 문서 수정
- [ ] CI/CD 변경

### 📷 관련 이미지
![스크린샷 2024-12-11 오후 6 41 35](https://github.com/user-attachments/assets/d6eb92ef-7fc5-42e6-a078-5ed65e1a8c0b)

![스크린샷 2024-12-11 오후 6 41 52](https://github.com/user-attachments/assets/2ee9e3e9-7c6b-4606-a525-bbc9d4fa54a1)

![스크린샷 2024-12-11 오후 6 42 32](https://github.com/user-attachments/assets/9f5d849d-a973-49d5-8e79-245ddb22dc17)


![스크린샷 2024-12-11 오후 6 42 20](https://github.com/user-attachments/assets/2c9dceb6-2636-4075-8a03-5289e8ebb315)

### 🚨 주의 사항
axios 인스턴스 적용은 다른 이슈를 통해 일괄적으로 개선할 예정입니다!

### ✅ 체크 리스트
- [ ] 코드 리뷰어가 이해할 수 있도록 주석 및 설명을 추가했습니다.
- [ ] 테스트를 추가하거나 업데이트했습니다.
- [ ] 문서를 업데이트했습니다 (해당 시).
- [x] 이 변경 사항이 다른 기능에 영향을 미치는지 확인했습니다.

dayaelee committed:
KAN-155 feat: 마이페이지 회원 탈퇴 기능 추가

dayaelee committed:
KAN-155 style: 마이페이지 회원 탈퇴 버튼 위치 조 정

konempty commented on src/pages/mypage/MyPage.jsx:
```diff
@@ -176,17 +179,42 @@ function MyPage(props) {
+  };
+
+  const accountDeactivation = (() => {
+    axios.get(`${API_URL}/mypage/delete`, { withCredentials: true })
```
저번에 말씀드린대로 axios는 공용 객체화 해서 쓰시면 좋을것 같아요~!

dayaelee replied:
해당 부분 수정하였습니다!

sall6550 reviewed (commented):
mypage 가로비율을 늘리면 좋을거같아요~

nuyeo commented on src/pages/mypage/MyPage.jsx:
```diff
@@ -256,7 +284,17 @@ function MyPage(props) {
+          <Popconfirm
+            title="주의"
+            description={<>탈퇴를 진행하시겠습니까?<br />
+              탈퇴 후 계정은 비활성화 상태로 전환되며, 데이터는 복구를 원하실 경우 일정 기간 동안 보관됩니다.</>}
```
문구가 적절하지 않은 것 같습니다.
그리고 처음 설계와 다르게 비활성화 과정을 추가하신 이유가 있나요? 비활성화 후 일정 기간 동안 로그인하지 않으면 탈퇴되는 로직일 경우에는 더 구체적인 안내가 필요할 것 같습니다.

dayaelee replied:
@nuyeo 일단 문구는 임시적으로 작성한거에요. soft-delete 방식으로 구현하는 걸로 이해하고 작성한거라 일단 취지에 맞는 문구라 생각해서 일단 저렇게 적어두긴 했어요. 그리고 사실 사용자 데이터 지운다곤 하지만 사용자가 작성한 피드백도 갖고 있는 거라 그런부분은 좀 애매하지 않나요?  제가 이해를 잘 못하고 있는거라면 혹시 제안하시는 적절한 문구있으시다면 추천부탁드립니다~

dayaelee replied:
@nuyeo 비활성화 후 일정 기간 동안 로그인하지 않으면 탈퇴되는 로직은 아닙니다 탈퇴 버튼을 누른경우에만 탈퇴 처리 됩니다~

nuyeo replied:
탈퇴 버튼을 누르는데 비활성화 상태로 전환된다고 나와서 비활성화가 되는지 탈퇴가 되는지 잘 구분이 안 됩니다
또 데이터가 일정 기간 보관된다고 나와있는데 일정 기간이 얼마만큼인지도 명시해야하지 않을까요?

nuyeo replied:
soft-delete 방식으로 구현한 이유는 `특정 블로그 후기`나 `피드백` 테이블 등 일부 테이블에서 `user_id`를 PK로 사용하고 있기 때문에 칼럼 자체를 날려버리면 에러가 발생할 수 있어, 필요한 데이터만 지우고 `deleted_at`에 탈퇴 시간을 저장해서 탈퇴한 사용자로 구분한다고 이해했는데 사용자에게 비활성화 상태로 전환된다는 안내를 할 필요는 없지 않을까요?

dayaelee replied:
아아 저부분은 메시지 창에 멘트만 저런거고 사실 기능은 요구사항 그대로 구현된 상태입니다

nuyeo replied:
그렇다면 멘트 수정이 필요할 것 같습니다!

dayaelee replied:
저는 '탈퇴를 진행하시겠습니까?' 정도로 넣으면 어떨까 싶은데 혹시 어떠신가요? 이외에 더 적절한 문구 아이디어 있으시면 의견 부탁드립니다!

nuyeo replied:
탈퇴 시 삭제할 데이터를 정하고 해당 데이터는 사라진다는 안내 문구를 넣으면 좋을 것 같아요

https://pin.it/2iteo4vDY

nuyeo replied:
피드백은 남겨놓는 게 좋을 것 같은데, 작성했던 특정 블로그 후기나 남겼던 좋아요는 남겨둘지 날릴지 팀원들과 의논이 필요할 것 같습니다

nuyeo replied:
답글 수정 내용이 새로고침 하기 전까지는 반영되지 않아서 더 코멘트 하실 부분이 있으면 새 댓글로 작성 부탁드립니다! 계속 수정하시면 확인이 어렵네요 ㅜㅜ

nuyeo replied:
@dayaelee 답글 수정 내용이 새로고침 하기 전까지는 반영되지 않아서 더 코멘트 하실 부분이 있으면 새 댓글로 작성 부탁드립니다! 계속 수정하시면 확인이 어렵네요 ㅜㅜ

dayaelee replied:
작성한 글에 오해의 소지가 있는것 같아서 수정하고 변경사항 있음 새로 코멘트 달겠습니다 !

dayaelee replied:
저도 피드백 부분은 남겨놔야 할것 같아요! 작성했던 특정 블로그 후기나 남겼던 좋아요를 남겨둘지에 대한 부분은 내일 1시 스크럼에서 안건 꺼내보고 해당 내용 기반으로 메세지 멘트 수정하겠습니다!

dayaelee replied:
멘트 수정 완료했습니다!

dayaelee commented:
> mypage 가로비율을 늘리면 좋을거같아요~

@sall6550 수정완료하였습니다!

dayaelee committed:
KAN-155 feat: 회원 탈퇴 API 요청 URL 추가 및 관련 요청 함수 수정
- requests.fetchUserDelete 경로 추가

dayaelee committed:
KAN-155 feat: axios 인스턴스 적용 및 프로필 컨테이너 비율 재설정, 회원탈퇴 멘트 수정

- axios 인스턴스를 활용하여 API 호출 방식 개선
- 프로필 컨테이너 및 내부 요소들의 반응형 비율 재설정 (min-width 및 flex-wrap 적용)
- Popconfirm 회원탈퇴 메시지 텍스트 수정
